4-Shaders-MoreAttributes/Window.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using OpenTK;
+using OpenTK.Graphics;
+using OpenTK.Graphics.OpenGL4;
+using OpenTK.Input;
+using LearnOpenTK.Common;
+
+namespace LearnOpenTK
+{
+    // In this project we will be assigning 3 colors to the triangle, one for vertex, but the output will be a mixture
+    // Of the three depending on how far away from the assignment point the color pixel is
+    // This comes down to something called fragment interpolation in the fragment shader
+    // When rendering a triangle the rasterization stage usually results in a lot more fragments than vertices originally specified. 
+    // The rasterizer then determines the positions of each of those fragments based on where they reside on the triangle shape
+    // Based on these positions, it interpolates all the fragment shader's input variables
+    public class Window : GameWindow
+    {
+
+        // We're assigning three different colors at the asscoiate vertex position
+        // Blue for the top, green for the bottom left and red for the bottom right
+        private readonly float[] _vertices =
+        {
+             // positions        // colors
+             0.5f, -0.5f, 0.0f,  1.0f, 0.0f, 0.0f,   // bottom right
+            -0.5f, -0.5f, 0.0f,  0.0f, 1.0f, 0.0f,   // bottom left
+             0.0f,  0.5f, 0.0f,  0.0f, 0.0f, 1.0f    // top 
+        };
+
+        private int _vertexBufferObject;
+
+        private int _vertexArrayObject;
+
+        private Shader _shader;
+
+        public Window(int width, int height, string title)
+            : base(width, height, GraphicsMode.Default, title)
+        {
+        }
+
+        // Now, we start initializing OpenGL.
+        protected override void OnLoad(EventArgs e)
+        {
+            GL.ClearColor(0.2f, 0.3f, 0.3f, 1.0f);
+
+            _vertexBufferObject = GL.GenBuffer();
+
+            GL.BindBuffer(BufferTarget.ArrayBuffer, _vertexBufferObject);
+            GL.BufferData(BufferTarget.ArrayBuffer, _vertices.Length * sizeof(float), _vertices, BufferUsageHint.StaticDraw);
+
+            _shader = new Shader("Shaders/shader.vert", "Shaders/shader.frag");
+
+            _shader.Use();
+
+            _vertexArrayObject = GL.GenVertexArray();
+            GL.BindVertexArray(_vertexArrayObject);
+
+
+            // We now need to account for 3 color values in the stride variable, so it has 
+            // Gone from 3 to 6
+            GL.VertexAttribPointer(0, 3, VertexAttribPointerType.Float, false, 6 * sizeof(float), 0);
+            GL.EnableVertexAttribArray(0);
+
+
+            // We create a new pointer for the color values much like the previous pointer we assign 6 
+            // In the stride value on top of which need to correctly set the offset to get the color values 
+            // we do by giving the amount of values there which is 3 and multiple them by their size
```

Comment:
The stride is the size in bytes so it is multiplied by the size of a single float. We dont assign 6 to the stride rather we tell it that it is 6 floats by doing 6*sizeof(float)